gloo/test/openssl_utils.cc

```diff
@@ -0,0 +1,94 @@
+#include "gloo/test/openssl_utils.h"
+
+#include <gloo/common/logging.h>
+
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include <memory>
+
+namespace gloo {
+namespace test {
+
+#define wrap_in_unique_ptr_prefix(NAME, TYPE, PREFIX)                         \
+auto NAME##_unique_ptr = [] {                                                 \
+  auto TYPE##_deleter = [](TYPE* x) { PREFIX##_free(x); };                    \
+  std::unique_ptr<TYPE, decltype(TYPE##_deleter)> x(                          \
+    PREFIX##_new(), TYPE##_deleter);                                          \
+  GLOO_ENFORCE(x.get(), #PREFIX"_new failed");                                \
+  return x;                                                                   \
+}();                                                                          \
+auto NAME = NAME##_unique_ptr.get()
+
+#define wrap_in_unique_ptr(NAME, TYPE) \
+  wrap_in_unique_ptr_prefix(NAME, TYPE, TYPE)
+
+void create_pkey_and_cert(const std::string& pkey_file_path,
+                          const std::string& cert_file_path) {
+  wrap_in_unique_ptr(pkey, EVP_PKEY);
+
+  wrap_in_unique_ptr_prefix(bn, BIGNUM, BN);
```

Comment:
IIUC, `BIGNUM` will be used as a type to create `BIGNUM*`. Where is `BIGNUM` defined?

Reply:
in `openssl/ossl_typ.h` via `openssl/x509.h`